crates/cargo-test-support/src/cross_compile.rs

```diff
@@ -180,6 +181,51 @@ rustup does not appear to be installed. Make sure that the appropriate
     panic!(message);
 }
 
+// Cache used for `native()` to avoid one process execution per call
+thread_local! {
+    static NATIVE_TARGET: RefCell<Option<&'static str>> = RefCell::new(None);
+}
+
+/// The arch of the test-running host.
+///
+/// Only use this function on tests that check `cross_compile::disabled`.
+pub fn native() -> &'static str {
+    NATIVE_TARGET.with(|t| {
+        if let Some(t) = *t.borrow() {
+            return t;
+        }
+        let out = String::from_utf8(
+            cargo::util::process("rustc")
```

Comment:
FWIW instead of executing `rustc -vV` could this use a build script to just forward the build scripts `env::var("TARGET")` to here?